src/client/clientmap.cpp

```diff
@@ -720,6 +737,12 @@ void ClientMap::renderMapShadows(video::IVideoDriver *driver,
 		}
 	}
 
+	// restore the driver material state 
+	video::SMaterial clean;
+	clean.BlendOperation = video::EBO_ADD;
+	driver->setMaterial(clean); // reset material to defaults
+	driver->draw3DLine(v3f(), v3f(), video::SColor(0));
```

Comment:
Why do we need draw a line here?

Reply:
This is to force irrlicht to actually update internal state when I set the material, otherwise there is no effect of the preceding lines.

Reply:
Add a comment for this, otherwise nobody will be able to remember.